cfg/CFG.h

```diff
@@ -120,26 +132,38 @@ class CFG final {
     static constexpr int ENSURE_BLOCK_OFFSET = 2;
     static constexpr int ELSE_BLOCK_OFFSET = 3;
 
-    UnorderedMap<core::LocalVariable, int> minLoops;
-    UnorderedMap<core::LocalVariable, int> maxLoopWrite;
-
     void sanityCheck(core::Context ctx);
 
     struct ReadsAndWrites {
-        std::vector<UnorderedSet<core::LocalVariable>> reads;
-        std::vector<UnorderedSet<core::LocalVariable>> writes;
+        std::vector<UnorderedSet<int>> readsSet;
+        std::vector<std::vector<int>> reads;
```

Comment:
Some places need fast readset lookup, which is most efficient as an `UnorderedSet`. Other places need fast iteration / merging of read sets, which is most efficient as a vector. Since `ReadsAndWrites` is not mutated once it's created, I stored them as both.